packages/squiggle-lang/src/value/boxed.ts

```diff
@@ -9,6 +11,32 @@ export type BoxedArgsType = {
   dateFormat?: string;
 };
 
+type BoxedArgsTypeName = keyof BoxedArgsType;
+
+const boxedArgsTypeNames: string[] = [
+  "name",
+  "description",
+  "showAs",
+  "numberFormat",
+  "dateFormat",
+];
+
+function convertToBoxedArgsTypeName(
+  key: string
+): result<BoxedArgsTypeName, string> {
```

Comment:
This is too minor to argue about, but for me it's another reminder that imperative code with early returns and type narrowing can be more readable than functional.

Here's how I'd do this:

```typescript
// `asserts` will narrow the type;
// see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions
function assertIsBoxedArgsTypeName(key: string): asserts key is BoxedArgsTypeName {
  if (!validKeys.has(key) {
    throw new REArgumentError(...);
  }
}

// [...]

  omitUsingStringKeys(keys: string[]) {
    const validKeys = keys.map((key) => {
      assertIsBoxedArgsTypeName(key);
      return key;
    });
    return this.omit(validKeys);
  }
```

Reply:
I think I like that improvement, feel free to change.